src/main/grammars/RustParser.bnf

```diff
@@ -983,6 +983,45 @@ MacroDefinition ::= MacroRulesInvocation identifier ItemMacroArg {
   elementTypeFactory = "org.rust.lang.core.stubs.StubImplementationsKt.factory"
 }
 
+// ident: an identifier. Examples: x; foo.
+// path: a qualified name. Example: T::SpecialA.
+// expr: an expression. Examples: 2 + 2; if true { 1 } else { 2 }; f(42).
+// ty: a type. Examples: i32; Vec<(char, String)>; &T.
+// pat: a pattern. Examples: Some(t); (17, 'a'); _.
+// stmt: a single statement. Example: let x = 3.
+// block: a brace-delimited sequence of statements and optionally an expression. Example: { log(error, "hi"); return 12; }.
+// item: an item. Examples: fn foo() { }; struct Bar;.
+// meta: a "meta item", as found in attributes. Example: cfg(target_os = "windows").
+// tt: a single token tree.
+
+MacroDefinitionPatternList ::= <<macro_args_delim (MacroDefinitionPattern ';'?)* >>
+MacroMatcher ::= MacroMatchingTree*
+MacroMatchingTree ::=  '(' MacroMatchingTrees ')'
+                 | '{' MacroMatchingTrees '}'
+                 | '[' MacroMatchingTrees ']'
+                 | (MacroSimpleMatching | MacroComplexMatching)+
+                 | << placeholderMacroToken >>+
+
+private MacroMatchingTrees ::=  MacroMatchingTree*
```

Comment:
I'd rather use `MacroMathingTree*` directly instead of a private rule: it's the same number of characters, and the meaning is more obvious! :)